tinywl/tinywl.c

```diff
@@ -552,6 +596,45 @@ static void server_cursor_frame(struct wl_listener *listener, void *data) {
 	wlr_seat_pointer_notify_frame(server->seat);
 }
 
+static void output_configure_scene(struct wlr_scene_node *node,
+		struct decoration_data *deco_data) {
+	if (!node->enabled) {
+		return;
+	}
+
+	struct tinywl_view *view = tinywl_view_addon_get(&node->addons, node);
+	if (view) {
+		deco_data = &view->deco_data;
+	}
+
+	if (node->type == WLR_SCENE_NODE_BUFFER) {
+		struct wlr_scene_buffer *buffer = wlr_scene_buffer_from_node(node);
+
+		struct wlr_scene_surface * scene_surface
+			= wlr_scene_surface_from_buffer(buffer);
+		struct wlr_xdg_surface *xdg_surface =
+			wlr_xdg_surface_from_wlr_surface(scene_surface->surface);
+
+		if (deco_data &&
+				xdg_surface &&
+				xdg_surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL) {
+			// TODO: Be able to set whole decoration_data instead of calling
+			// each individually?
+			wlr_scene_buffer_set_opacity(buffer, deco_data->alpha);
+
+			if (!wlr_surface_is_subsurface(xdg_surface->surface)) {
+				wlr_scene_buffer_set_corner_radius(buffer, deco_data->corner_radius);
+			}
```

Comment:
What's your opinion on this? Maybe add a `wlr_scene_buffer_set_decoration_data` while also keeping the other functions?